### ISSUE TITLE
[BUG] MeloTTS fails to initialize when using GPU mode

### ISSUE THREAD

**Issue by ibewackofreak** [hkjarral/Asterisk-AI-Voice-Agent#195]
## Bug Description
When selecting GPU mode instead of CPU mode for MeloTTS, the tts engine will fail to initialize due to cuda versions.

## Steps to Reproduce
1.  Select MeloTTS for TTS
2.  Change device from CPU to GPU and apply changes
3.  Check docker logs for errors

## Expected Behavior
2026-02-07 16:39:44 -0500 INFO root: 🎙️ TTS backend: MeloTTS (voice=EN-BR, device=cuda, speed=1.0)
2026-02-07 16:39:44 -0500 INFO root: 🎙️ MELOTTS - Initializing (voice=EN-BR, device=cuda, speed=1.0)
config.json: 3.49kB [00:00, 17.7MB/s]
/opt/venv/lib/python3.11/site-packages/torch/nn/utils/weight_norm.py:144: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
checkpoint.pth: 100% 208M/208M [00:02<00:00, 90.1MB/s]
2026-02-07 16:39:47 -0500 INFO root: ✅ MELOTTS - TTS initialized successfully
2026-02-07 16:39:47 -0500 INFO root: ✅ TTS backend: MeloTTS initialized (44100Hz native)
2026-02-07 16:39:47 -0500 INFO root: ✅ All models loaded successfully for MVP pipeline
2026-02-07 16:39:47 -0500 INFO root: 🚀 Enhanced Local AI Server started on ws://127.0.0.1:8765

## Actual Behavior
2026-02-07 16:32:02 -0500 INFO root: 🎙️ TTS backend: MeloTTS (voice=EN-BR, device=cuda, speed=1.0)
2026-02-07 16:32:02 -0500 INFO root: 🎙️ MELOTTS - Initializing (voice=EN-BR, device=cuda, speed=1.0)
2026-02-07 16:32:02 -0500 ERROR root: ❌ MELOTTS - Failed to initialize:
2026-02-07 16:32:02 -0500 ERROR root: ❌ Failed to initialize MeloTTS backend: Failed to initialize MeloTTS
2026-02-07 16:32:02 -0500 WARNING root: ⚠️ Local AI started in degraded mode (failed: tts).

## Environment
- **Version**: v6.0.0
- **OS**: Ubuntu 24.04.3
- **Docker Version**: CE 29.2.1
- **Asterisk Version**: 22.8.2
- **Configuration**: local
- **Transport**: ExternalMedia RTP

## Logs

```
/opt/venv/lib/python3.11/site-packages/huggingface_hub/file_download.py:949: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
tokenizer_config.json: 100% 251/251 [00:00<00:00, 3.44MB/s]
vocab.txt: 231kB [00:00, 33.1MB/s]
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/appuser/nltk_data...
[nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.
[nltk_data] Downloading package cmudict to /home/appuser/nltk_data...
[nltk_data]   Unzipping corpora/cmudict.zip.
tokenizer_config.json: 100% 48.0/48.0 [00:00<00:00, 662kB/s]
config.json: 100% 570/570 [00:00<00:00, 6.36MB/s]
vocab.txt: 100% 232k/232k [00:00<00:00, 42.9MB/s]
tokenizer.json: 100% 466k/466k [00:00<00:00, 17.2MB/s]
tokenizer_config.json: 100% 48.0/48.0 [00:00<00:00, 727kB/s]
config.json: 100% 625/625 [00:00<00:00, 8.24MB/s]
vocab.txt: 100% 872k/872k [00:00<00:00, 25.1MB/s]
tokenizer.json: 100% 1.72M/1.72M [00:00<00:00, 35.4MB/s]
tokenizer_config.json: 100% 80.0/80.0 [00:00<00:00, 1.28MB/s]
config.json: 100% 725/725 [00:00<00:00, 12.8MB/s]
vocab.txt: 344kB [00:00, 135MB/s]
tokenizer_config.json: 100% 83.0/83.0 [00:00<00:00, 1.28MB/s]
config.json: 100% 420/420 [00:00<00:00, 6.70MB/s]
vocab.txt: 227kB [00:00, 154MB/s]
tokenizer_config.json: 100% 310/310 [00:00<00:00, 5.63MB/s]
config.json: 100% 650/650 [00:00<00:00, 11.1MB/s]
vocab.txt: 248kB [00:00, 141MB/s]
tokenizer.json: 486kB [00:00, 161MB/s]
special_tokens_map.json: 100% 134/134 [00:00<00:00, 2.45MB/s]
2026-02-07 16:31:36 -0500 INFO root: 🚀 Initializing enhanced AI models for MVP...
2026-02-07 16:31:36 -0500 INFO root: 🎤 STT backend: Faster-Whisper (model=medium, device=cpu, compute=float32)
2026-02-07 16:31:36 -0500 INFO root: 🎤 FASTER-WHISPER - Loading model=medium device=cpu compute=float32
config.json: 2.26kB [00:00, 4.70MB/s]
vocabulary.txt: 460kB [00:00, 24.0MB/s]
tokenizer.json: 2.20MB [00:00, 96.1MB/s]
model.bin: 100% 1.53G/1.53G [00:13<00:00, 113MB/s]
2026-02-07 16:31:51 -0500 INFO root: ✅ FASTER-WHISPER - Model loaded successfully
2026-02-07 16:31:51 -0500 INFO root: ✅ STT backend: Faster-Whisper initialized
llama_context: n_ctx_per_seq (768) < n_ctx_train (32768) -- the full capacity of the model will not be utilized
2026-02-07 16:31:52 -0500 INFO root: ✅ LLM model loaded: /app/models/llm/qwen2.5-3b-instruct-q4_k_m.gguf (CPU only)
2026-02-07 16:31:52 -0500 INFO root: 📊 LLM Config: ctx=768, threads=6, batch=128, max_tokens=64, temp=0.4, gpu_layers=0
2026-02-07 16:31:52 -0500 INFO root: 🧪 LLM WARMUP START - Running startup latency check (prompt_tokens=39 raw_tokens=39 model=qwen2.5-3b-instruct-q4_k_m.gguf ctx=768 batch=128 max_tokens<=32)
2026-02-07 16:31:57 -0500 INFO root: ⏳ LLM WARMUP - In progress (~5s elapsed, model=qwen2.5-3b-instruct-q4_k_m.gguf ctx=768 batch=128)
2026-02-07 16:32:02 -0500 INFO root: 🤖 LLM STARTUP LATENCY - 9963.34 ms (prompt_tokens=39 raw_tokens=39 truncated=False)
2026-02-07 16:32:02 -0500 INFO root: 🎙️ TTS backend: MeloTTS (voice=EN-BR, device=cuda, speed=1.0)
2026-02-07 16:32:02 -0500 INFO root: 🎙️ MELOTTS - Initializing (voice=EN-BR, device=cuda, speed=1.0)
2026-02-07 16:32:02 -0500 ERROR root: ❌ MELOTTS - Failed to initialize:
2026-02-07 16:32:02 -0500 ERROR root: ❌ Failed to initialize MeloTTS backend: Failed to initialize MeloTTS
2026-02-07 16:32:02 -0500 WARNING root: ⚠️ Local AI started in degraded mode (failed: tts).
2026-02-07 16:32:02 -0500 INFO root: 🚀 Enhanced Local AI Server started on ws://127.0.0.1:8765
2026-02-07 16:32:02 -0500 INFO root: 📋 Pipeline: ExternalMedia (8kHz) → STT (16kHz) → LLM → TTS (8kHz µ-law) | Supports selective STT/TTS modes
2026-02-07 16:32:06 -0500 INFO root: 🔌 Client connected: ('127.0.0.1', 52878)
2026-02-07 16:32:29 -0500 INFO root: 🔌 Client connected: ('127.0.0.1', 56600)
2026-02-07 16:32:29 -0500 INFO root: 🔐 WS AUTH - Authenticated session call_id=unknown
```

## Diagnostics (Recommended)

These make issues actionable and help us update the Supported Platforms matrix:

```bash
========================================
AAVA Pre-flight Checks
========================================

✓ Architecture: x86_64
✓ OS: ubuntu 24.04 (debian family)
✓ Docker: 29.2.1
✓ Docker Compose: 5.0.2
✓ Docker Buildx: 0.31
✓ Media directory (host): /opt/Asterisk-AI-Voice-Agent/asterisk_media/ai-generated
ℹ   Container path: /mnt/asterisk_media/ai-generated
✓ Data directory (host): /opt/Asterisk-AI-Voice-Agent/data
✓ Call history DB: writable (SQLite test passed)
✓ Models directory (host): /opt/Asterisk-AI-Voice-Agent/models
✓ call_history.db ownership: OK (writable by containers)
✓ .env file exists
ℹ   Tip: For local_only pipeline, no API keys needed!
✓ Asterisk binary: Asterisk 22.8.2
✓ Asterisk config: /etc/asterisk
✓ FreePBX: FW Console - Incredible PBX Utility 17.0.25
✓ Asterisk UID:GID = 999:1002
✓ NVIDIA GPU detected: NVIDIA GeForce RTX 3050 (6144 MiB)
✓ nvidia-container-toolkit installed
ℹ Testing Docker GPU passthrough...
✓ Docker GPU passthrough working
ℹ   Verified with image: nvidia/cuda:12.0.0-base-ubi8
ℹ
ℹ   GPU will be detected by Setup Wizard automatically (via GPU_AVAILABLE in .env)
ℹ
ℹ   To use GPU for LLM inference (optional, faster responses):
ℹ     1. Set LOCAL_LLM_GPU_LAYERS=-1 in .env
ℹ     2. Start local_ai_server with GPU override:
ℹ        docker compose -p asterisk-ai-voice-agent -f docker-compose.yml -f docker-compose.gpu.yml up -d --build local_ai_server
⚠ Port 3003 already in use (Admin UI port)

========================================
Pre-flight Summary
========================================
Warnings (1):
  ⚠ Port 3003 already in use (Admin UI port)

Auto-fixable issues (run with --apply-fixes):
  sudo chgrp 1002 /opt/Asterisk-AI-Voice-Agent/asterisk_media/ai-generated
  sudo chmod 2775 /opt/Asterisk-AI-Voice-Agent/asterisk_media/ai-generated  # setgid for group inheritance
  sudo chgrp 1002 /opt/Asterisk-AI-Voice-Agent/data
  sudo chmod 2775 /opt/Asterisk-AI-Voice-Agent/data  # setgid for group inheritance (SQLite call history)
  sudo ln -sfn /opt/Asterisk-AI-Voice-Agent/asterisk_media/ai-generated /var/lib/asterisk/sounds/ai-generated  # allow Asterisk to serve generated audio

Checks passed with warnings.
```

If the `agent` CLI is not installed:
```
Asterisk AI Voice Agent CLI
Version:    v6.0.0
Built:      2026-02-07T06:31:21Z
Repository: https://github.com/hkjarral/Asterisk-AI-Voice-Agent
```

Note for forks: replace `hkjarral/Asterisk-AI-Voice-Agent` in the URL above with your fork owner/repo.

Then attach diagnostics:
```
Asterisk AI Voice Agent - agent check (v6.0.0)
══════════════════════════════════════════
Timestamp: 2026-02-07T16:46:01-05:00
CLI Version: v6.0.0
Build: 2026-02-07T06:31:21Z

[1/18] Host...                    ✅ Host context collected
      hostname=talkland.stevesworld.org
kernel=6.8.0-94-generic
[2/18] Docker CLI...              ✅ docker found
[3/18] Docker Daemon...           ✅ running
      server_version=29.2.1
[4/18] Docker Compose...          ✅ available
      version=5.0.2
[5/18] Container ai_engine...     ✅ running
      image=asterisk-ai-voice-agent-ai-engine:latest
status=running
network_mode=host
[6/18] Network Mode...            ✅ host networking
      NetworkMode=host
[7/18] Mounts...                  ✅ required mounts present
      /app/data <- /opt/Asterisk-AI-Voice-Agent/data (rw=true)
/mnt/asterisk_media <- /opt/Asterisk-AI-Voice-Agent/asterisk_media (rw=true)
[8/18] Container local_ai_server... ✅ running
      image=asterisk-ai-voice-agent-local-ai-server:latest
status=running
network_mode=host
health=healthy
[9/18] Local AI Models...         ✅ models mount looks ok
      host_models=/opt/Asterisk-AI-Voice-Agent/models
ai_engine_models=/app/models <- /opt/Asterisk-AI-Voice-Agent/models
local_ai_server_models=/app/models <- /opt/Asterisk-AI-Voice-Agent/models
/app/models exists=true
/app/models/stt exists=true
/app/models/tts exists=true
/app/models/llm exists=true
/app/models/kroko exists=true
[10/18] In-Container Paths...      ✅ paths look ok
      /app/data exists=true writable=true
/mnt/asterisk_media exists=true writable=true
/mnt/asterisk_media/ai-generated exists=true writable=true
[11/18] Call History DB...         ✅ writable (SQLite test passed)
      canonical_path=/app/data/call_history.db
[12/18] Config...                  ✅ loaded
      app_name=asterisk-ai-voice-agent
audio_transport=externalmedia
active_pipeline=(none)
downstream_mode=stream
[13/18] Env...                     ✅ loaded (values redacted by design)
      ASTERISK_HOST=127.0.0.1
ASTERISK_ARI_PORT=8088
ASTERISK_ARI_SCHEME=http
ASTERISK_APP_NAME=(unset)
CALL_HISTORY_DB_PATH=data/call_history.db
EXTERNAL_MEDIA_ADVERTISE_HOST=(unset)
AUDIOSOCKET_ADVERTISE_HOST=(unset)
LOCAL_WS_URL=ws://127.0.0.1:8765
[14/18] Transport Compatibility... ✅ looks compatible
[15/18] Advertise Hosts...         ✅ looks ok
[16/18] ARI...                     ✅ reachable
      url=http://127.0.0.1:8088
asterisk_version=22.8.2
ari_app_registered=true
[17/18] Dialplan...                ✅ ARI shows app is registered
      expected_stasis_app=asterisk-ai-voice-agent
[18/18] Internet/DNS...            ✅ reachable (best-effort)
      deb.debian.org:443 dns=true tcp=true ms=17
api.openai.com:443 dns=true tcp=true ms=11
api.deepgram.com:443 dns=true tcp=true ms=66

══════════════════════════════════════════
Summary
PASS 18  WARN 0  FAIL 0  SKIP 0  (18 total)
Overall: PASS (system looks healthy)
```

If this is a call-specific issue, also attach:
```bash
# Most recent call
agent rca

# Or a specific call_id (preferred)
agent rca --call <call_id>
```

Optional (for automation / parsing):
```bash
agent check --json
agent rca --json
```

## Configuration

```yaml
# config/ai-agent.yaml (redacted)
```

## Checklist
- [ X] I have searched existing issues for duplicates
- [ X] I have redacted all sensitive information (API keys, passwords)
- [ X] I have included relevant logs
- [ X] I have specified my environment details

**Comment by hkjarral** [comment]
Resolved in `v6.1.1` (`2026-02-10`) via MeloTTS CUDA guard + runtime CPU fallback.

Relevant changes:
- `fix(gpu): enforce cuda guard and add runtime cpu fallback for stt/tts` (`0387353c`)
- Runtime GPU probe/indicators for clearer host/runtime compatibility

What changed:
- If MeloTTS is set to `cuda` and CUDA runtime is incompatible, startup now falls back to CPU instead of failing TTS initialization.

Upgrade/rebuild steps:
```bash
git fetch --tags
git checkout v6.1.1
docker compose up -d --build --force-recreate
```

Closing as fixed in `v6.1.1`. Reopen if still reproducible post-upgrade with logs.
